File: csharp/976.largest-perimeter-triangle.cs
/*
 * @lc app=leetcode id=976 lang=csharp
 *
 * [976] Largest Perimeter Triangle
 */

using System;

// @lc code=start
public class Solution {
    public int LargestPerimeter(int[] A) {
        Array.Sort(A);
        for (int i = A.Length - 1; i > 1; --i) {
            int l = A[i], m = A[i - 1], s = A[i - 2];
            if (m + s > l) {
                return l + s + m;
            }
        }
        return 0;
    }
}
// @lc code=end

File: java/976.largest-perimeter-triangle.java
/*
 * @lc app=leetcode id=976 lang=java
 *
 * [976] Largest Perimeter Triangle
 */

import java.util.Arrays;

// @lc code=start

class Solution {
    public int largestPerimeter(int[] nums) {
        Arrays.sort(nums);
        int n = nums.length;
        for (int i = n - 1; i >= 2; --i) {
            int l = nums[i], m = nums[i - 1], s = nums[i - 2];
            if (m + s > l) {
                return m + s + l;
            }
        }
        return 0;
    }
}
// @lc code=end

